### PR TITLE
Add threading build test to all.sh

### PR DESCRIPTION
**Pull request by andresag01**

### CODE REVIEW

File: tests/scripts/all.sh [hunk 1]
```diff
@@ -482,6 +482,13 @@ make test
 cd "$MBEDTLS_ROOT_DIR"
 rm -rf "$OUT_OF_SOURCE_DIR"
 
+msg "build: threading"
+cleanup
+cp "$CONFIG_H" "$CONFIG_BAK"
+scripts/config.pl set MBEDTLS_THREADING_C
+scripts/config.pl set MBEDTLS_THREADING_PTHREAD
+PTHREAD=1 make
```
Comment (gilles-peskine-arm) [inline tests/scripts/all.sh]: This needs to be changed to `make PTHREAD=1` to work in keep-going mode.

Why test the build but not the tests?

**Comment by sbutcher-arm** [comment]
I think I'd prefer to get #1243 merged first, and see this rebased on top of that, rather than slow down #1243.

cc: @mpg

File: tests/scripts/all.sh (hunk 1 above)
Comment (andresag01) [reply tests/scripts/all.sh]: This was a while ago, but as far as I remember, there are no dedicated tests for concurrency in the library. For this reason, we created [this PR](https://github.com/ARMmbed/mbedtls/pull/658) to test an Mbed TLS server (programs/ssl/ssl_pthread_server.c) managing multiple client connections.

**Comment by daverodgman** [comment]
Closing (@AndrzejKurek is currently looking at threading work).

**Pull request closed without merging**